### ISSUE TITLE
need the ability to include title for c3 charts

### ISSUE THREAD

**Issue by madasuk** [c3js/c3#1021]
Hi, Is it possible to provide the feature of including title for the c3 charts ?  I have come across donut: { title: 'Title'} at http://c3js.org/reference.html for donut chart title but could not able find any reference to include title for other charts.

**Comment by davidpoore** [comment]
I added a pull request for this feature, but the build for my pull request seems stuck in pending mode, don't know what to do about that : /

**Comment by davidpoore** [comment]
Nevermind, finally went green

**Comment by davidpoore** [comment]
I've also opened another issue with a pull request to allow users to add header backgrounds - don't know if this is something you would also be interested in: #1031

**Comment by madasuk** [comment]
Thanks, any estimation by when this feature would be available?

**Comment by davidpoore** [comment]
@aendrew said he would look at the pull request tomorrow, probably

**Comment by aendrew** [comment]
Hi, apologies for the tremendous delay. Looking at it now.

**Comment by masayuki0812** [comment]
#1025 has been merged. This new feature will be released in the next version `v0.4.11`. Thanks!

**Comment by masayuki0812** [comment]
I added `title.position` and `title.padding` to enable us to set its position more easily. Instead, `title.x` and `title.y` has been removed because setting title position by `x` and `y` is not intuitive and `title.padding` seems easier to use.
Btw, this commit is based on @davidpoore 's work #1025 . Thank you again :)

**Comment by crudbug** [comment]
:100:

**Comment by aendrew** [comment]
@masayuki0812 Given this is now in 0.4.11-RC1, can we close this and #1031 yet?

**Comment by arjenm** [comment]
My pie-chart disappeared when I added a padding-element to the title, only when all four values where entered did it show. I.e. with padding:{} or padding:{top:0} the pie-chart disappeared. But with padding:{top:0, bottom:0, left:0, right:0} it did work.

I'd expect the other values to keep their defaults of zero, rather than do something eratic.

To make the title fit directly to the top of the chart, I had to use top: -4, which is kinda counter-intuitive. Although that might stem from the line-height (and thus the bounding box).
